src/components/utils/CalciteExpandToggle.tsx

```diff
@@ -0,0 +1,67 @@
+import { FunctionalComponent, h } from "@stencil/core";
+import { getElementDir } from "./dom";
+import { CalciteLayout } from "../interfaces";
+
+interface CalciteExpandToggleProps {
+  expanded: boolean;
+  textExpand: string;
+  textCollapse: string;
+  el: HTMLElement;
+  layout: CalciteLayout;
+  toggleExpand: () => void;
+}
+
+const ICONS = {
+  chevronsLeft: "chevrons-left",
+  chevronsRight: "chevrons-right"
+};
+
+function getClosestShellLayout(el: HTMLElement): CalciteLayout {
+  const shellNode = el.closest("calcite-shell-panel");
+
+  if (!shellNode) {
+    return;
+  }
+
+  return shellNode.layout;
+}
+
+export function setActionTextEnabled({
```

Comment:
Suggestion: API-wise, what do you think about renaming this to `toggleChildActionText({ parent, enabled})` or something similar?